packages/autotask-client/src/bin.ts

```diff
@@ -1,41 +1,162 @@
+ * @param command The command to validate.
+ */
+function mandatoryParamGuard(command: Command) {
+  switch (command) {
+    case 'update-code':
+      if (!argv[3] || !argv[4]) {
+        printUsage();
+        process.exit(1);
+      }
+      break;
+    // Same requirements for now
+    case 'tail-runs':
+    case 'execute-run':
+      if (!argv[3]) {
+        printUsage();
+        process.exit(1);
+      }
+      break;
+    default:
+      printUsage();
+      process.exit(1);
+  }
+}
+
+/* -------------------------------- Commands -------------------------------- */
+
+/**
+ * Utilizes the Autotask API to update the code of a given autotask.
+ */
 async function updateCode() {
   const autotaskId = argv[3];
   const path = argv[4];
-  const { API_KEY: apiKey, API_SECRET: apiSecret } = process.env;
-
-  if (autotaskId.length !== 36) throw new Error(`invalid autotask id '${autotaskId}'`);
-  if (!existsSync(path)) throw new Error(`path ${path} does not exist`);
-  if (!apiKey || !apiSecret) throw new Error(`API_KEY or API_SECRET env vars are missing`);
+  try {
+    validateId(autotaskId);
+    validatePath(path);
 
-  const client = new AutotaskClient({ apiKey, apiSecret });
-  console.error(`Uploading code for autotask ${autotaskId} from ${path}...`);
-  await client.updateCodeFromFolder(autotaskId, path);
+    const client = initClient();
+    console.error(`Uploading code for autotask ${autotaskId} from ${path}...`);
+    await client.updateCodeFromFolder(autotaskId, path);
+  } catch (error) {
+    const err = error as Error | AxiosError;
+    console.error(`Error updating Autotask code: ${err.message}`);
+    process.exit(1);
+  }
 }
 
-async function main() {
-  if (argv[2] !== 'update-code' || !argv[3] || !argv[4]) {
-    printUsage();
-    return;
+/**
+ * Utilizes the Autotask API to poll for new runs and print them out.
+ */
+async function tailRuns() {
+  const autotaskId = argv[3];
+  const logSavePath = argv[4];
+  let logSaveDest: string | undefined;
+  let stream: WriteStream | undefined;
+  if (logSavePath) {
+    validatePath(logSavePath);
+    logSaveDest = `${logSavePath}/${autotaskId}-runs.log`;
+    stream = createWriteStream(logSaveDest, { flags: 'a' });
   }
 
   try {
-    await updateCode();
-  } catch (err) {
-    console.error(`Error updating Autotask code: ${err.message}`);
+    validateId(autotaskId);
+    const client = initClient();
+    output(`\nPolling latest runs of autotask '${autotaskId}'...\n`, stream);
+    // Poll autotask runs every 2 seconds and if there are new runs, get run details and print them out.
+    let lastRun: AutotaskRunListItemResponse | undefined;
+    while (true) {
+      const newRuns = await client.listAutotaskRuns(autotaskId);
+      // If cached last run id has changed
+      if (newRuns.items[0]?.autotaskRunId !== lastRun?.autotaskRunId) {
+        lastRun = newRuns.items[0]; // cache new last run to avoid duplicates.
+
+        if (lastRun.status !== 'success') {
```

Comment:
Heads up that a run has other terminal statuses besides `success`!

Reply:
Hope [this is](https://github.com/OpenZeppelin/defender-client/pull/99/commits/978372de470cc934a8ed17de6aab362ae352c69c) closer to your liking :)

I've dropped the custom `output`, minimized the verbosity, and followed (loosely) the frontend handling of the run statuses.

Reply:
LGTM, thanks!